tsec-libsodium/src/main/scala/tsec/libsodium/kx/kx.scala

```diff
@@ -0,0 +1,20 @@
+package tsec.libsodium
+
+import tsec.libsodium.cipher.SodiumKey
+
+package object kx {
+
+  final case class SodiumSharedKeyPair[A](receive: SodiumKey[A], send: SodiumKey[A])
+
+  case object KeySessionError extends Exception with Product with Serializable {
+    def cause: String = "KeySession generation Error"
+    override def fillInStackTrace(): Throwable = this
```

Comment:
is there a reason we're not using `scala.util.control.NoStackTrace`?

Reply:
didn't know about this

Reply:
@hrhino maybe `NoStackTrace` belongs in a different PR, for _all_ of our errs.